core/App/ehsm_napi.cpp

```diff
@@ -1155,15 +1160,60 @@ char *NAPI_RA_HANDSHAKE_MSG0(const char *p_msg0)
     log_d("***NAPI_RA_HANDSHAKE_MSG0 start.");
 
     log_d("msg0: \n %s", p_msg0);
-    std::string challenge_base64 = "challenge_base64";
-    std::string ga_base64 = "ga_base64";
-
     RetJsonObj retJsonObj;
+    ehsm_status_t ret = EH_OK;
+    sgx_ra_msg1_t *p_msg1;
+
+    memset(&p_msg1, 0, sizeof(p_msg1));
+
+    std::string challenge;
+    if (p_msg0 != nullptr)
+    {
+        Json::Value msg0_json;
+        Json::CharReaderBuilder builder;
+        Json::CharReader *reader(builder.newCharReader());
+        std::string err;
+        std::string response = p_msg0;
+        if (reader->parse(response.c_str(), response.c_str() + response.size(), &msg0_json, &err))
+        {
+            challenge = msg0_json["challenge"].asString();
+        }
+    }
+    else
+    {
+        retJsonObj.setCode(retJsonObj.CODE_BAD_REQUEST);
+        retJsonObj.setMessage("paramter invalid.");
+        goto out;
+    }
+    if(challenge.empty()){
```

Comment:
suggest to refine the layout as:
if ( !p_msg0 ||  challenge.empty() || xxxxx....) {
    return or goto out; 
}

the common logic....

Reply:
It will be change